### PR TITLE
Move test specific option setting to the test file

### PR DESCRIPTION
**Pull request by chendave**
Setting of "metricsRecorder" is only needed for testcase.

Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/runtime/framework_test.go [hunk 1]
```diff
@@ -1756,6 +1756,13 @@ func TestPermitPlugins(t *testing.T) {
 	}
 }
 
+// withMetricsRecorder set metricsRecorder for the scheduling frameworkImpl, only used in tests.
```
Comment (hex108) [inline pkg/scheduler/framework/runtime/framework_test.go:1759]: `only used in tests.` is not needed now.

Comment (chendave) [reply pkg/scheduler/framework/runtime/framework_test.go:1759]: sorry, I cannot get your point here. do you mean the this function is not needed in tests? or update the comment to say this is not needed anymore?

Comment (hex108) [reply pkg/scheduler/framework/runtime/framework_test.go:1759]: Update the comment. :)

`withMetricsRecorder set metricsRecorder for the scheduling frameworkImpl`

Comment (chendave) [reply pkg/scheduler/framework/runtime/framework_test.go]: Done, thanks.

**Comment by hex108** [comment]
thanks

/lgtm

**Pull request merged**